### PR TITLE
link , scraper for question 

### PR DESCRIPTION
**Pull request by Eyobyb**
find link , scrape data , summerize data and integrate the scraped and summarized data back into the question as a reference

### CODE REVIEW

File: apps/slackbot/utils.py [hunk 1]
```diff
@@ -14,6 +22,108 @@ def load_files(files: List[str]) -> List[Document]:
...
+
+def get_base_url(link):
+    parsed_url = urlparse(link)
+    base_url = f"{parsed_url.scheme}://{parsed_url.netloc}"
+    return base_url
+
+
+def get_link_from_slack_client_conversation(data):
+    links = []
+    for item in data:
```
Comment (saminegash) [inline apps/slackbot/utils.py:56]: what is the nested for loop here @Eyobyb can we kind of try optimising this

**Review by saminegash — CHANGES_REQUESTED** [review]

File: apps/slackbot/utils.py (hunk 1 above)
Comment (Eyobyb) [reply apps/slackbot/utils.py:56]: the slack message structure is like that .. no way of joining about it

Comment (saminegash) [reply apps/slackbot/utils.py:56]: There are almost 4 nested for loops here man

Comment (Eyobyb) [reply apps/slackbot/utils.py:56]: [ ##1 loop
  {
    'client_msg_id': '960d41d8-2718-4ce5-b5d5-ff66aa09af24',
    'type': 'message',
    'text': '<@U05HDFV64AU> what does <https://www.reuters.com/world/africa/ethiopia-declares-state-emergency-following-militia-clashes-2023-08-04/> talk about ethiopia',
    'user': 'U05GGHG7T1C',
    'ts': '1691420055.909099',
    'blocks': [
 ##2 loop
      {
        'type': 'rich_text',
        'block_id': 'CCqFe',
        'elements': [
 ##3 loop
          {
            'type': 'rich_text_section',
            'elements': [
 ##4 loop
              {
                'type': 'user',
                'user_id': 'U05HDFV64AU'
              },
              {
                'type': 'text',
                'text': ' what does '
              },
              {
                'type': 'link',
                'url': 'https://www.reuters.com/world/africa/ethiopia-declares-state-emergency-following-militia-clashes-2023-08-04/'
              },
              {
                'type': 'text',
                'text': ' talk about ethiopia'
              }
            ]
          }
        ]
      }
    ],]

Comment (Eyobyb) [reply apps/slackbot/utils.py:56]: you can see it from this structure in order to get the link you have to go down for nested arrays

Comment (saminegash) [reply apps/slackbot/utils.py:56]: Okay let me look into this and try optimizing it then thanks will look into this

**Review by saminegash — APPROVED** [review]

**Pull request merged**